added collapsible widget

sagardspeed2 opened the pull request:
## Describe your changes
Added a collapsible widget with functionality of accordion.
Features:
Item template
can pass custom icons
can change header and content background color

### Screenshots [Optional]

## Issue ticket number and link

Closes #
closes #297 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [x] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

evshi commented on apps/kitchen-sink/src/ensemble/screens/widgets.yaml:
```diff
@@ -20,6 +20,29 @@ View:
+            value:
+              - "1"
+            expandIconPosition: "end"
+            items:
```
This should take a generic list of children like Column/Row

sagardspeed2 replied:
@evshi I think here we do not need of passing children like column/rows,
because here we used prebuilt collapsible widget of antd design.

here is the reference,
https://ant.design/components/collapse

evshi commented on apps/kitchen-sink/src/ensemble/screens/widgets.yaml:
```diff
@@ -20,6 +20,29 @@ View:
+            text: "## Collapse"
+        - Collapsible:
+            id: collapseWidget
+            value:
+              - "1"
+            expandIconPosition: "end"
```
How about supporting either a header widget and allow them to customize the header entirely OR support a title and icon

sagardspeed2 replied:
I already added a support for that, user can pass any widget as a header for each collapsible item.
for example,
```js
items:
  - key: "1"
    label:
      Text:
        text: Here is my label 1
    children: hiii
  - key: "2"
    label: Here is my label 2
    children:
      Text:
        text: Here is my value 2
```

here label represent the header of collapsible and children represent the content which will be shown when the item will be collapse

evshi commented on packages/runtime/src/widgets/Collapsible.tsx:
```diff
@@ -0,0 +1,84 @@
+} & EnsembleWidgetProps;
+
+export const Collapsible: React.FC<CollapsibleProps> = (props) => {
+  const [activeValue, setActiveValue] = useState<string[]>(props.value);
```
Does this represent whether it's collapsed or not?

sagardspeed2 replied:
This collapsible widget takes an array of labels and children as input. Based on this array, it creates a list of collapsibles. If the activeValue state has any value, it indicates that one or multiple collapsibles from that array are collapsed.

evshi replied:
I see. We will need to collapse items individually and read their value. The invokable method that is registered should be a callback that takes the index and value to make it easier to use `(index: int, value: string) => void` over passing an entire array.

evshi reviewed (commented):
@anserwaseem can you please review and see if it meets the requirements you need?

anserwaseem commented:
> @anserwaseem can you please review and see if it meets the requirements you need?

@sagardspeed2 would need few more things:
1. `item-template` 
2. ability to give my own icon
3. styles for collapsible head (I need to change its background colour) - low priority

evshi commented:
@sagardspeed2 something like this?

```yaml
Collapsibe:
  item-template:
     data
     name: item
     template:
         CollapsibleItem:
             label ${item.title}
             key: ${item.id}
             item:
                Text:
                    text: ${item.value}
```

sagardspeed2 commented:
> > @anserwaseem can you please review and see if it meets the requirements you need?
> 
> @sagardspeed2 would need few more things:
> 
> 1. `item-template`
> 2. ability to give my own icon
> 3. styles for collapsible head (I need to change its background colour) - low priority

Added item-template, ability to give custom icon, also added ability to change the collapsible header and content background color

evshi reviewed (approved):
GEFN to unblock @anserwaseem. @sagardspeed2 please add a changeset
